src/main/java/uk/gov/hmcts/reform/sscs/servicebus/TopicConsumer.java

```diff
@@ -49,8 +50,7 @@ private void processMessageWithRetry(String message, int retry) {
             processMessage(message);
         } catch (Exception e) {
             if (retry > maxRetryAttempts) {
-                // retried and now unrecoverable. Catch to remove it from the queue.
-                log.error(format("Caught unknown unrecoverable error %s", e.getMessage()), e);
+                throw new UnrecoverableError(String.format("Caught unknown unrecoverable error %s", e.getMessage()), e);
```

Comment:
Is this correct? Will this allow an errored case to leave the queue? or is there a risk it will just keep retrying and then end up on dead letter queue?

Reply:
the issue with the log.error is that it's causing functional tests to fail silently. What I mean is the functional tests might fail and you will still see it green because we are catching the exception but not throwing just logging as an error
on the other hand, we are not doing anything with those unknown unrecoverable errors for the time being, apart from logging an error. Therefore, this is not going to change that. (edite

Reply:
We don't want to throw any exceptions as the JMS retry will kick in, and the message will persist in the dead letter queue (if configured). We should log the error so that Ops get alerted.

Reply:
revert it as it was before. Cheers!